hal_st/stm32fxxx/DmaStm.cpp

```diff
@@ -57,151 +58,219 @@ namespace hal
-            {
-                DMA2_Channel1,
-                DMA2_Channel2,
-                DMA2_Channel3,
-                DMA2_Channel4,
-                DMA2_Channel5,
-                DMA2_Channel6,
+        const std::array DmaChannelDma2
+        {
+            DMA2_Channel1,
+            DMA2_Channel2,
+            DMA2_Channel3,
+            DMA2_Channel4,
+            DMA2_Channel5,
+            DMA2_Channel6,
 #if defined(DMA2_Channel7)
-                DMA2_Channel7,
+            DMA2_Channel7,
 #endif
 #if defined(DMA2_Channel8)
-                DMA2_Channel8,
-#endif
-            },
+            DMA2_Channel8,
 #endif
         };
-        const std::array dmaIrq
+#endif
+
+#if defined(DMA2_Channel1)
+        const std::array<decltype(DmaChannelDma1), 2> DmaChannel { DmaChannelDma1, DmaChannelDma2 };
+#else
+        const std::array<decltype(DmaChannelDma1), 1> DmaChannel { DmaChannelDma1 };
+#endif
+
+        const std::array dmaIrqDma1
         {
-            std::array
-            {
-                DMA1_Channel1_IRQn,
-                DMA1_Channel2_IRQn,
-                DMA1_Channel3_IRQn,
-                DMA1_Channel4_IRQn,
-                DMA1_Channel5_IRQn,
-                DMA1_Channel6_IRQn,
+#if defined(STM32WBA)
+            GPDMA1_Channel0_IRQn,
+            GPDMA1_Channel1_IRQn,
+            GPDMA1_Channel2_IRQn,
+            GPDMA1_Channel3_IRQn,
+            GPDMA1_Channel4_IRQn,
+            GPDMA1_Channel5_IRQn,
+            GPDMA1_Channel6_IRQn,
+            GPDMA1_Channel7_IRQn,
+#else
+            DMA1_Channel1_IRQn,
+            DMA1_Channel2_IRQn,
+            DMA1_Channel3_IRQn,
+            DMA1_Channel4_IRQn,
+            DMA1_Channel5_IRQn,
+            DMA1_Channel6_IRQn,
+#endif
 #if defined(DMA1_Channel7)
-                DMA1_Channel7_IRQn,
+            DMA1_Channel7_IRQn,
 #endif
 #if defined(DMA1_Channel8)
-                DMA1_Channel8_IRQn,
+            DMA1_Channel8_IRQn,
 #endif
-            },
+        };
 #if defined(DMA2_Channel1)
-            std::array
-            {
-                DMA2_Channel1_IRQn,
-                DMA2_Channel2_IRQn,
-                DMA2_Channel3_IRQn,
-                DMA2_Channel4_IRQn,
-                DMA2_Channel5_IRQn,
-                DMA2_Channel6_IRQn,
+        const std::array dmaIrqDma2
+        {
+            DMA2_Channel1_IRQn,
+            DMA2_Channel2_IRQn,
+            DMA2_Channel3_IRQn,
+            DMA2_Channel4_IRQn,
+            DMA2_Channel5_IRQn,
+            DMA2_Channel6_IRQn,
 #if defined(DMA2_Channel7)
-                DMA2_Channel7_IRQn,
+            DMA2_Channel7_IRQn,
 #endif
 #if defined(DMA2_Channel8)
-                DMA2_Channel8_IRQn,
-#endif
-            },
+            DMA2_Channel8_IRQn,
 #endif
         };
+#endif
+
+#if defined(DMA2_Channel1)
+        const std::array<decltype(dmaIrqDma1), 2> dmaIrq { dmaIrqDma1, dmaIrqDma2 };
+#else
+        const std::array<decltype(dmaIrqDma1), 1> dmaIrq { dmaIrqDma1 };
+#endif
```

Comment:
I am not _positive_ on if this would compile:

```suggestion
        const std::array dmaIrq
        {
            dmaIrqDma1,
#if defined(DMA2_Channel1)
            dmaIrqDma2
#endif
         };
```

Reply:
Unfortunately this original idea does not work with just one channel, that is why I've changed to this new approach. You can check it here: https://onlinegdb.com/2QRM-lOvH.